src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -978,6 +981,91 @@ export class ResultsViewPageStore {
         }
     });
 
+
+    readonly filteredAlterationsDataByUnflattenedOQLLine = remoteData<
+    IQueriedMergedTrackCaseData[]
+>({
+    await: () => [
+        this.filteredAlterations,
+        this.annotatedMolecularData,
+        this.selectedMolecularProfiles,
+        this.defaultOQLQuery,
+        this.samples,
+        this.patients
+    ],
+    invoke: () => {
+        const data = [...(this.filteredAlterations.result!), ...(this.annotatedMolecularData.result!)];
+        const accessorsInstance = new accessors(this.selectedMolecularProfiles.result!);
+        const defaultOQLQuery = this.defaultOQLQuery.result!;
+        const samples = this.samples.result!;
+        const patients = this.patients.result!;
+
+        if (this.oqlQuery.trim() === '') {
+            return Promise.resolve([]);
+        } else {
+            const filteredAlterationsByOQLLine: UnflattenedOQLLineFilterOutput<AnnotatedExtendedAlteration>[] = (
+                filterCBioPortalWebServiceDataByUnflattenedOQLLine(
+                    this.oqlQuery,
+                    data,
+                    accessorsInstance,
+                    defaultOQLQuery
+                )
+            );
+
+            return Promise.resolve(filteredAlterationsByOQLLine.map(
+                (oql) => ({
+                    cases: groupDataByCase(oql, samples, patients),
+                    oql,
+                    list: filterSubQueryData(
+                        oql, defaultOQLQuery,
+                        data, accessorsInstance,
+                        samples, patients
+                    )
+                })
+            ));
+        }
+    }
+});
+
+    readonly isSampleAlteredMap = remoteData({
+        await: () => [this.filteredAlterationsDataByUnflattenedOQLLine, this.samples],
+        invoke: async() => {          
+            const result : {[x: string]: boolean[]} = {};
+            this.filteredAlterationsDataByUnflattenedOQLLine.result!.forEach((element, key) => {
```

Comment:
there is a lot of logic here inside this invoke.  it should be unit tested, which will require that you put the function into a "pure" helper function.   what you could do is make this into this.filterAlterationDataByUnglattendOQLine.result.map(   someTestableExternalFunction(this.samples.result) );

Reply:
Ok! I will add unit tests for this function!